src/utils/range.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Creates an array of numbers (positive and/or negative) progressing from
+ * start up to, but not including, end. A step of -1 is used if a negative start
+ * is specified without an end or step. If end is not specified, it's set to
+ * start with start then set to 0.
+ */
+export const range = (start: number, end?: number, step?: number): number[] => {
+   if (end === undefined) {
+      end = start;
+      start = 0;
+   }
+   step = step ?? (start < end ? 1 : -1);
+
+   if (step === 0) {
+      const length = Math.max(end - start, 0);
+
+      return Array(length).fill(start);
+   }
```

Comment:
We've traditionally used underscore instead of lodash. I'm rather surprised by the behavior differences:

Underscore:

```js
_.range(2, 5)
(3) [2, 3, 4]
_.range(2, 5, undefined)
(3) [2, 3, 4]
_.range(2, 5, null)
(3) [2, 3, 4]
_.range(2, 5, 0)
(3) [2, 3, 4]
_.range(2, 5, false)
(3) [2, 3, 4]
```

Lodash

```js
_.range(2, 5)
(3) [2, 3, 4]
_.range(2, 5, undefined)
(3) [2, 3, 4]
_.range(2, 5, null)
(3) [2, 2, 2]
_.range(2, 5, 0)
(3) [2, 2, 2]
_.range(2, 5, false)
(3) [2, 2, 2]
``` 

Can we opt for the underscore behavior? We don't typically lump all falsey values together with `if (!step)`. But in this case, it seems more standard to have all falsey values behave the same.